modules/core/location/farm_location.module

```diff
@@ -44,6 +44,18 @@ function farm_location_entity_base_field_info(EntityTypeInterface $entity_type)
   }
 }
 
+/**
+ * Implements hook_entity_base_field_info_alter().
+ */
+function farm_location_entity_base_field_info_alter(&$fields, EntityTypeInterface $entity_type) {
+  /** @var \Drupal\field\Entity\FieldConfig[] $fields */
+
+  // Prevent creating circular asset location.
+  if ($entity_type->id() == 'log' && !empty($fields['asset'])) {
```

Comment:
Should this also do the `is_movement` check?

Reply:
I suppose we could include `&& !empty($fields['is_movement'])` here too, but all log types have that field, so it wouldn't add much (unless that changes in the future I suppose). Although to be fair, all log types also have an `asset` field, so that check is equally unnecessary here I suppose.

We do check the SPECIFIC value of `is_movement` in `CircularAssetLocationConstraintValidator` (below): https://github.com/farmOS/farmOS/pull/568/files#diff-a423c1299c8e264f45b3abda4e862ea21b300de6946241c5618168fdb587d0b9R54